stable/elasticsearch-exporter/values.yaml

```diff
@@ -11,6 +11,10 @@ image:
   tag: 1.0.2
   pullPolicy: IfNotPresent
 
+securityContext:
```

Comment:
Maybe state that you should comment in order to disable.

Usually this is done by a enabled flag but it does not work this way here unless we explicitely get runAsNonRoot runAsUser in deployment.yaml. What do you think?

Reply:
What is needed in the securityContext depends on the version of the image you use and the flavor of Kubernetes. I think it is good to have it completely configurable.
I added a comment in the value file.

Reply:
i double-checked how other charts are doing it. It seems to be standard to use a flag for enabling/disabling configuration snippets (as @desaintmartin mentioned already), e.g.

```
{{- if .Values.securityContext.enabled }}
    securityContext:
        runAsNonRoot: {{ .Values.securityContext.runAsNonRoot }}
        runAsUser: {{ .Values.securityContext.runAsUser }}
       ...
{{- end }}
```

would this be an option?

Reply:
ok, will do that.

Reply:
@svenmueller this is an option.
However, with your example it would not be possible to set runAsNonRoot without setting runAsUser. 
I could do:
```yaml
{{- if .Values.securityContext.enabled }}
    securityContext:
        {{- if .Values.securityContext.runAsNonRoot }}
        runAsNonRoot: {{ .Values.securityContext.runAsNonRoot }}
        {{- end}}
        {{- if .Values.securityContext.runAsUser }}
        runAsUser: {{ .Values.securityContext.runAsUser }}
         {{- end }}
       ...
{{- end }}
```
but it seems quite heavy, not sure it is really worth it.

@svenmueller , @desaintmartin what do you think?

Reply:
in that case, i would prefer you current solution.

Reply:
Ok, thanks @svenmueller .

The lgtm label is still needed to merge this PR.

Reply:
Who can put the lgtm label (my understanding of the process is that it comes before the approve label)?

Reply:
I've just realized something.
Actually, if you comment the whole section in your own custom values.yaml, it will take values from the default values.yaml (this file!) and it will be still used. Same if you comment `runAsNonRoot` for example.
This is why the `enabled` flags are preferred.
But maybe we can workaround this by setting `securityContext: {}` or something like this.

Reply:
@desaintmartin You're right, it is not good. :(.

Initially, I wanted to let the securityContext configuration 100% configurable. However, it is not what other stable charts in this repository do.
I searched this repository for securityContext.enabled and I propose to use the same pattern as they use in the [mongodb chart](https://github.com/helm/charts/blob/30b5c13ae1cad13655c0e3f43c7f4cf525e76944/stable/mongodb/templates/deployment-standalone.yaml#L66). It is not as configurable as I initially wanted it to be but it covers the two cases I am aware of, Kubernetes and OpenShift.

Reply:
I agree, I don't like this pattern (any new parameter requires a PR) but this might be the simplest way for now.